### ISSUE TITLE
[BUG] `utf8_encode()` has been deprecated.

### ISSUE THREAD

**Issue by rosiel** [Islandora/islandora#988]
PHP Note: https://php.watch/versions/8.2/utf8_encode-utf8_decode-deprecated

**What steps does it take to reproduce the issue?**

* Caught by [CI](https://github.com/Islandora/islandora/actions/runs/6640203344/job/18040153209). 

* When does this issue occur?
When running CI on PHP 8.2.

* Where does it occur?
https://github.com/Islandora/islandora/blob/2.x/modules/islandora_text_extraction/src/Plugin/Field/FieldFormatter/OcrTextFormatter.php#L135-L137

## Resolution

`mbstring`, `iconv` and `intl` are suggested as replacements, but I don't know enough to suggest which one (or combination) to use, or whether we have access to these extensions in the default deployment.

**Comment by adam-vessey** [comment]
Given the `utf8_encode()` call is predicated on a call to `mb_detect_encoding()`, which seems to be from `mbstring`, would seem to point towards `mbstring` as the shortest path, already expected to be present?

Instead of the current

```php
    if (mb_detect_encoding($contents) != 'UTF-8') {
      $contents = utf8_encode($contents);
    }
```

Something like:

```php
    $detected_encoding = mb_detect_encoding($contents);
    if ($detected_encoding != 'UTF-8') {
      $contents = mb_convert_encoding($contents, 'UTF-8', $detected_encoding);
    }
```

?

---

Also, revisiting Drupal's requirements proper, it requires `mbstring` already: https://www.drupal.org/docs/getting-started/system-requirements/php-requirements#s-mbstring-

**Comment by rosiel** [comment]
fixed as part of #987
